arcgis-ios-sdk-samples/Layers/ArcGIS tiled layer (tile cache)/TilePackagesListViewController.swift

```diff
@@ -33,69 +33,78 @@ class TilePackagesListViewController: UITableViewController {
     }
     
     func fetchTilePackages() {
-        self.bundleTPKPaths = Bundle.main.paths(forResourcesOfType: "tpk", inDirectory: nil)
-        self.tableView.reloadData()
+        // Fetch URLs for tile packages in the main bundle.
+        if let urls = Bundle.main.urls(forResourcesWithExtension: "tpk", subdirectory: nil) {
+            bundleTilePackageURLs = urls
+        }
         
+        // Fetch URLs for tile packages in the documents directory.
         let documentDirectoryURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
-        let subpaths = FileManager.default.subpaths(atPath: documentDirectoryURL.path)
-        
-        let predicate = NSPredicate(format: "SELF MATCHES %@", ".*tpk$")
-        let tpks = subpaths?.filter { (objc) -> Bool in
-            return predicate.evaluate(with: objc)
-        }
-        self.documentTPKPaths = tpks?.map { (name: String) -> String in
-            return documentDirectoryURL.appendingPathComponent(name).path
+        if let subpaths = FileManager.default.subpaths(atPath: documentDirectoryURL.path) {
+            let predicate = NSPredicate(format: "SELF MATCHES %@", ".*tpk$")
+            documentTilePacakgeURLs = subpaths.lazy
+                .filter { predicate.evaluate(with: $0) }
+                .map { documentDirectoryURL.appendingPathComponent($0) }
         }
     }
     
-    // MARK: - UITableViewDataSource
+    /// A section in the table view.
+    ///
+    /// - bundle: The section showing tile packages in the bundle.
+    /// - documentsDirectory: The section showing tile packages in the documents
+    /// directory.
+    enum Section: CaseIterable {
+        case bundle
+        case documentsDirectory
+    }
     
+    /// Returns the `URL` corresponding to the row at the given index path.
+    ///
+    /// - Parameter indexPath: An index path.
+    /// - Returns: A `URL`.
+    func urlForRow(at indexPath: IndexPath) -> URL {
+        switch Section.allCases[indexPath.section] {
+        case .bundle:
+            return bundleTilePackageURLs[indexPath.row]
+        case .documentsDirectory:
+            return documentTilePacakgeURLs[indexPath.row]
+        }
+    }
+}
+
+extension TilePackagesListViewController /* UITableViewDataSource */ {
     override func numberOfSections(in tableView: UITableView) -> Int {
         return 2
```

Comment:
Since you went through the trouble of making `enum Section` why not return `Section.allCases.count` on line `77`?

This would be more congruent with making the table sections configuration driven.

Reply:
Thanks. Meant to do that.